geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureWriter.scala

```diff
@@ -173,29 +179,30 @@ class ModifyAccumuloFeatureWriter(featureType: SimpleFeatureType,
   // index or table. This list is configured to match the
   // version of the datastore (i.e. single table vs catalog
   // table + index tables)
-  val removers: List[SimpleFeature => Unit] = {
+  val removers: List[FeatureWriterFn] = {
     val stTable = dataStore.getSpatioTemporalIdxTableName(featureType)
     val stWriter = List(SpatioTemporalTable.removeSpatioTemporalIdx(multiBWWriter.getBatchWriter(stTable), indexEncoder))
 
     val rowIdPrefix = org.locationtech.geomesa.core.index.getTableSharingPrefix(featureType)
 
-    val attrWriters: List[SimpleFeature => Unit] =
+    val attrWriters =
       if (dataStore.catalogTableFormat(featureType)) {
         val attrTable = dataStore.getAttrIdxTableName(featureType)
         val recTable = dataStore.getRecordTableForType(featureType)
 
         List(
-          AttributeTable.removeAttrIdx(multiBWWriter.getBatchWriter(attrTable), featureType, indexedAttributes, visibility, rowIdPrefix),
-          RecordTable.recordDeleter(multiBWWriter.getBatchWriter(recTable), encoder, visibility, rowIdPrefix))
+          AttributeTable.removeAttrIdx(multiBWWriter.getBatchWriter(attrTable), featureType, indexedAttributes, rowIdPrefix),
+          RecordTable.recordDeleter(multiBWWriter.getBatchWriter(recTable), encoder, rowIdPrefix))
       } else {
         List.empty
       }
     stWriter ::: attrWriters
   }
 
+  import scala.collection.JavaConversions._
   override def remove() =
     if (original != null) {
-      removers.foreach { r => r(original) }
+      removers.foreach { r => r(original, original.getUserData.getOrElse(security.SecurityUtils.FEATURE_VISIBILITY, visibility).asInstanceOf[String]) }
```

Comment:
i don't think that we currently encode user data with the feature - i suspect that updating a feature will result in the old row not getting deleted because of that... Could you add a unit test to verify one way or the other?